File: packages/shared/types/graph.ts
/**
 * Shared Graph visualization types used across mobile app and API responses.
 */

export type NodeType = 'drug' | 'enzyme' | 'metabolite' | 'receptor';

export interface GraphNode {
  id: string;
  label: string;
  type: NodeType;
  // Type-specific properties
  class?: string;       // drug
  family?: string;      // enzyme
  active?: boolean;     // metabolite
  effect?: string;      // receptor
  // Layout coordinates (computed server-side for performance)
  x?: number;
  y?: number;
}

export interface GraphEdge {
  source: string;
  target: string;
  type: string;
  strength?: string;
}

export interface PatientGraph {
  nodes: GraphNode[];
  edges: GraphEdge[];
}

File: packages/api/models/graph.py
"""
Cascadex API — Graph Visualization Pydantic Models.
"""

from typing import Optional

from pydantic import BaseModel


class GraphNode(BaseModel):
    id: str
    label: str
    type: str  # 'drug', 'enzyme', 'metabolite', 'receptor'
    # Type-specific fields
    drug_class: Optional[str] = None
    family: Optional[str] = None
    active: Optional[bool] = None
    effect: Optional[str] = None


class GraphEdge(BaseModel):
    source: str
    target: str
    type: str
    strength: Optional[str] = None


class PatientGraph(BaseModel):
    nodes: list[GraphNode] = []
    edges: list[GraphEdge] = []
